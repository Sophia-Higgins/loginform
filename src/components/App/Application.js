import React, { useEffect, useState } from "react";
import { API_BASE_URL } from "../../constants/apiConstants";
import axios from "axios";

const Application = () => {
  const [message, setMessage] = useState("");

  const payload = {
    username: "Sophia",
  };

  useEffect(() => {
    axios.post(API_BASE_URL + "/greeting", payload).then((response) => {
      setMessage(response.data);
    });
  });

  return (
    <div className="body">
      <br />
      <h1 className="display-1 text-center text-primary">{message}</h1>
    </div>
  );
};

export default Application;
