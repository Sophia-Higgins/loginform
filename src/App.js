import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import PrivateRoute from "./utils/PrivateRoute";
import Header from "./components/Header/Header";
import LoginForm from "./components/LoginForm/LoginForm";
import Home from "./components/Home/Home";
import Account from "./components/Account/Account";
import Application from "./components/App/Application";
import Payment from "./components/Payment/Payment";

function App() {
  return (
    <Router>
      <div className="App">
        <Header />
        <Switch>
          <Route path="/" exact={true}>
            <div className="container d-flex align-items-center flex-column">
              <LoginForm />
            </div>
          </Route>
          <Route path="/login">
            <div className="container d-flex align-items-center flex-column">
              <LoginForm />
            </div>
          </Route>
          <PrivateRoute path="/home">
            <Home />
          </PrivateRoute>
          <PrivateRoute path="/account">
            <Account />
          </PrivateRoute>
          <PrivateRoute path="/application">
            <Application />
          </PrivateRoute>
          <PrivateRoute path="/payment">
            <Payment />
          </PrivateRoute>
        </Switch>
      </div>
    </Router>
  );
}

export default App;
