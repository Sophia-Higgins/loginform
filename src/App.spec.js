import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import App from "./App";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<App />);
});

describe("<App />", () => {
  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
