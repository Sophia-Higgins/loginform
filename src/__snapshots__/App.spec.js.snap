// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match the snapshot 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <Header />
    <Switch>
      <Route
        exact={true}
        path="/"
      >
        <div
          className="container d-flex align-items-center flex-column"
        >
          <LoginForm />
        </div>
      </Route>
      <Route
        path="/login"
      >
        <div
          className="container d-flex align-items-center flex-column"
        >
          <LoginForm />
        </div>
      </Route>
      <PrivateRoute
        path="/home"
      >
        <Home />
      </PrivateRoute>
    </Switch>
  </div>
</BrowserRouter>
`;
